packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -51,6 +51,12 @@ function getMessage(e: ValidationError): string {
       );
 }
 
+function getEmptyErrorMessage(path: string, value: any, expectedType: string) {
+  return `Empty value ${stringify(
+    value
+  )} for ${path} - Expected a non-empty value of type ${expectedType}.`;
+}
+
```

Comment:
I think I would rename this to `getEmptyStringErrorMessage`, drop the `expectedType` parameter, and use a more specific error message like:

```
Empty string ${stringify(value)} for ${path} - Expected a non-empty string.
```

Wdyt?